components/tools/CardData.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useRef, useState } from 'react';
+import TextTruncate from 'react-text-truncate';
+
+import type { VisibleDataListType } from '@/types/components/tools/ToolDataType';
+
+import InfoIcon from '../icons/InfoIcon';
+
+interface CardDataProps {
+  visible: VisibleDataListType;
+  heading: string;
+  data: string;
+  read: boolean;
+  setRead: React.Dispatch<React.SetStateAction<boolean>>;
+  setVisible: React.Dispatch<React.SetStateAction<VisibleDataListType>>;
+  type: keyof VisibleDataListType;
+  className?: string;
+};
+
+/**
+ * @description This component displays Card.
+ *
+ * @param {SelectTagsProps} props - The props for the Cards Data component.
+ * @param {string} props.className - Additional CSS classes for the component.
+ * @param {VisibleDataListType} props.visible - Visibility status for different types.
+ * @param {string} props.heading - The heading text.
+ * @param {string} props.data - The data to be displayed.
+ * @param {boolean} props.read - Read status.
+ * @param {React.Dispatch<React.SetStateAction<boolean>>} props.setRead - Function to set read status.
+ * @param {React.Dispatch<React.SetStateAction<VisibleDataListType>>} props.setVisible - Function to set visibility status.
+ * @param {string} props.type - Type of the card data.
+ */
+export const CardData = ({
+  visible,
+  heading,
+  data,
+  read,
+  setRead,
+  setVisible,
+  type,
+  className = ''
+}: CardDataProps) => {
+  const [outsideClick, setOutsideClick] = useState<boolean>(true);
+  const [description, setShowDescription] = useState<boolean>(false);
+  const initial = {
+    lang: false,
+    tech: false,
+    category: false,
+    pricing: false,
+    ownership: false
+  };
+  const domNode = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    const divHeight = domNode.current?.offsetHeight || 0;
+    const numberOfLines = divHeight / 20;
+
+    if (numberOfLines > 3) {
+      setShowDescription(true);
+    } else {
+      setShowDescription(false);
+    }
+  }, [visible]);
```

Comment:
Specify in the comment, on why we are using useEffect here.

Reply:
Done